airflow/contrib/hooks/gcp_transfer_hook.py

```diff
@@ -56,52 +85,282 @@ def get_conn(self):
+        30 days after soft delete.
+
+        :param job_name: Name of the job to be deleted
+        :type job_name: str
+        :param project_id: The ID of the GCP project.
+        :type project_id: str
+        :rtype: None
+        """
+
+        return self.get_conn().transferJobs().patch(
+            jobName=job_name,
+            body={
+                'project_id': project_id,
+                'transfer_job': {
+                    'status': GcpTransferJobsStatus.DELETED
+                },
+                'update_transfer_job_field_mask': 'status'
+            },
+        ).execute(num_retries=NUM_RETRIES)
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def cancel_transfer_operation(self, operation_name):
+        """
+        Cancels an transfer operation in Google Storage Transfer Service.
+
+        :param operation_name: Name of the transfer operation.
+        :type operation_name: str
+        :rtype: None
+        """
+        self.get_conn()\
+            .transferOperations()\
+            .cancel(name=operation_name)\
+            .execute(num_retries=NUM_RETRIES)
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def get_transfer_operation(self, operation_name):
+        """
+        Gets an transfer operation in Google Storage Transfer Service.
+
+        :param operation_name: Name of the transfer operation.
+        :type operation_name: str
+        :return: transfer operation
+        :rtype: dict
+        """
+        return self.get_conn()\
+            .transferOperations()\
+            .get(name=operation_name)\
+            .execute(num_retries=NUM_RETRIES)
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def list_transfer_operations(self, filter):
+        """
+        Gets an transfer operation in Google Storage Transfer Service.
+
+        :param filter: A request filter, as described in
+            https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferJobs/list#body.QUERY_PARAMETERS.filter
+            With one additional improvement:
+
+            * project_id is optional if you have a project id defined
+              in the connection
+              See: :ref:`connection-type-GCP`
+
+        :type filter: dict
+        :return: transfer operation
+        :rtype: list[dict]
+        """
+        conn = self.get_conn()
+
+        filter = self._append_project_id(filter, 'filter project_id parameter')
+
+        operations = []
+
+        request = conn.transferOperations().list(
+            name=TRANSFER_OPERATIONS,
+            filter=json.dumps(filter)
+        )
+
+        while request is not None:
+            response = request.execute(num_retries=NUM_RETRIES)
+            if 'operations' in response:
+                operations.extend(response['operations'])
+
+            request = conn.transferOperations().list_next(
+                previous_request=request,
+                previous_response=response)
```

Comment:
A similar block of code appears several time. Is there a chance to extract it to a method and parameterize?

Reply:
Such a snippet of code also occurs elsewhere in the project. I have a parked issue related to list, list_next: https://trello.com/c/Vj2ujgAD  
I will try then to think about your comment.  I will try then to think about your comment when I solve it.